.github/workflows/cmake.yml

```diff
@@ -0,0 +1,62 @@
+name: CMake
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install boost
+        uses: MarkusJx/install-boost@v2.4.1
+        id: install-boost
+        with:
+          boost_version: 1.73.0
+          
+      - name: Setup NuGet
+        if: ${{ matrix.os=='windows-latest' }}
+        uses: NuGet/setup-nuget@v1.1.1
+        with:
+          nuget-version: '5.x'
+        
+      - name: Install NetworkDirect
+        if: ${{ matrix.os=='windows-latest' }}
+        run: mkdir -p objects/nuget && cd objects/nuget && nuget install NetworkDirect -Version 2.0.1
```

Comment:
Nice to have: A comment that says that if you update the version, go update `CONTIRBUTING.md`.